src/components/Main.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { TodoList } from './TodoList';
+import { Todo } from '../types/Todo';
+import { useTodos } from '../hooks/useTodo';
+
+export const Main: React.FC = () => {
+  const { todos, setTodos } = useTodos();
+  const [allCompleted, setAllComleted] = useState(true);
+
+  const toggleTodosActive = () => {
+    if (todos.length) {
+      const completedTodos: Todo[] = todos.map(todo => {
+        return { ...todo, completed: allCompleted };
+      });
+
+      setTodos(completedTodos);
+
+      setAllComleted(!allCompleted);
+    }
+  };
+
+  return (
+    <section className="main">
+      {!!todos.length && (
+        <>
+          <input
+            type="checkbox"
+            id="toggle-all"
+            className="toggle-all"
+            data-cy="toggleAll"
+            onClick={() => toggleTodosActive()}
```

Comment:
Will works same but less callbacks
```suggestion
            onClick={toggleTodosActive}
```